### PR TITLE
Modify Vagrantfile to use Debian 9

### PR DESCRIPTION
**Pull request by skylerbunny**
Upgrades the Vagrant .box file to use Debian 9.1 (and associated libraries), and Postgresql 9.6.

Wheels for Weasyl specific libraries are not available yet, so build must be performed as it exists in the Makefile (i.e., with no-binary). If this pull is approved, back end wheel builds will be upgraded as well as production workers are upgraded.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -120,9 +120,9 @@ make install-libweasyl upgrade-db
 SCRIPT
 
 Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
-  config.vm.box = "weasyl-debian81.box"
-  config.vm.box_url = "http://deploy.weasyldev.com/weasyl-debian81.box"
-  config.vm.box_download_checksum = "34592e65ebd4753d6f74a54b019e36d1ce006010cb4f03ed8ec131824f45ff9b"
+  config.vm.box = "weasyl-debian91.box"
+  config.vm.box_url = "http://deploy.weasyldev.com/weasyl-debian91.box"
```
Comment (charmander) [inline Vagrantfile]: This redirects to HTTPS; could you update it to specify `https:`?

Comment (skylerbunny) [reply Vagrantfile]: Done!

**Comment by kfkitsune** [comment]
Well, Vagrant doesn't want to bring up the box. More specifically, I see it's running if I launch the Virtualbox GUI, and it is at the login screen. Though it doesn't login at all. 

The previous version of the box came up properly, but after purging the previous file and attempting to issue a ``vagrant up``, it produces the following:

```
kyra@kitty:~/git/weasylDev/weasyl$ vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'weasyl-debian91.box'...
==> default: Matching MAC address for NAT networking...
==> default: Setting the name of the VM: weasyl_default_1506227089245_4204
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 8443 => 8443 (adapter 1)
    default: 22 => 2222 (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection timeout. Retrying...
    default: Warning: Connection timeout. Retrying...
    default: Warning: Connection timeout. Retrying...
[... more of the above ...]
Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period.

If you look above, you should be able to see the error(s) that
Vagrant had when attempting to connect to the machine. These errors
are usually good hints as to what may be wrong.

If you're using a custom box, make sure that networking is properly
working and you're able to connect to the machine. It is a common
problem that networking isn't setup properly in these boxes.
Verify that authentication configurations are also setup properly,
as well.

If the box appears to be booting properly, you may want to increase
the timeout ("config.vm.boot_timeout") value.
```

**Comment by skylerbunny** [comment]
I think I see what might be happening. For convenience, I've installed the virtualbox guest extensions (they do muck inside a vbox guest with filesystems, networking, etc.), but they're version specific. This means that virtualbox wants to be >= 5.1.28. I'm also using the newest vagrant, >= 2.0.x. Please upgrade to both if you haven't and give this another try.

**Comment by skylerbunny** [comment]
@kfkitsune I've changed the README to reflect the Vagrant and Virtualbox minimum version requirements.

**Pull request merged**